src/core/middleware/theme.ts

```diff
@@ -100,14 +137,32 @@ export const theme = factory(
 				icache.set(css, theme, false);
 				return theme;
 			},
-			set(css: Theme): void {
-				const currentTheme = injector.get<Injector<Theme | undefined>>(INJECTED_THEME_KEY);
-				if (currentTheme) {
-					currentTheme.set(css);
+			variant() {
+				let theme;
+				const { theme: themeProp } = properties();
```

Comment:
The theme property should be either the explicitly passed theme or the injected theme already, so can just use the property?